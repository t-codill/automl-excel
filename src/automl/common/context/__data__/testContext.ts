import { PageNames } from "../../PageNames";
import { Logger } from "../../utils/logger";
import { IWorkspaceContextProps, WorkspaceError } from "../IWorkspaceProps";
import { WorkspaceFlight } from "../WorkspaceFlight";

export function getLogCustomEventSpy(): jest.SpyInstance<ReturnType<Logger["logCustomEvent"]>> {
    const spy = jest.spyOn(Logger.prototype, "logCustomEvent");
    spy.mockImplementation(jest.fn());
    return spy;
}
let pageName: PageNames = PageNames.Unknown;
export const testContext: IWorkspaceContextProps = {
    get pageName(): PageNames { return pageName; },
    get theme(): string { return "light"; },
    discoverUrls: {},
    flight: new WorkspaceFlight(""),
    location: "eastus",
    resourceGroupName: "testResource",
    subscriptionId: "00000000-0000-0000-0000-000000000000",
    logger: new Logger("test"),
    getToken: () => "testToken",
    setPageName: (value: PageNames) => { pageName = value; },
    workspaceName: "testWorkSpace",
    setNavigationBarButtons(): void { return; },
    onError: (err: WorkspaceError) => { throw err; },
    clearErrors: () => { return; },
    setHeader: () => { return; },
    setLoading: () => { return; }
};
