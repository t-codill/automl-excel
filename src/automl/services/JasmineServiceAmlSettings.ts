import { AzureMachineLearningWorkspacesModels } from "@azure/arm-machinelearningservices";
import { IWorkspaceProps } from "../common/context/IWorkspaceProps";
import { AdvancedSetting } from "./AdvancedSetting";
import { MetricOperation } from "./constants/MetricOperation";

const maxEnsembleIterations = 15;
export const getAmlSettings = (
    experimentName: string,
    compute: AzureMachineLearningWorkspacesModels.ComputeResource,
    props: IWorkspaceProps,
    advancedSettingParams: AdvancedSetting
) => {
    if (!compute.properties) {
        throw new Error(`${compute.name} is not a valid compute`);
    }

    let ensembleIterations: number | null;
    let enableEnsembling: boolean;
    const iterations = advancedSettingParams.maxIteration;
    if (iterations > 2) {
        enableEnsembling = true;
        ensembleIterations = Math.min(iterations, maxEnsembleIterations);
    }
    else {
        enableEnsembling = false;
        ensembleIterations = null;
    }

    let jobType = advancedSettingParams.jobType;
    if (jobType === "forecasting") {
        jobType = "regression";
    }

    return {
        name: experimentName,
        path: `./sample_projects/${experimentName}`,
        subscription_id: props.subscriptionId,
        resource_group: props.resourceGroupName,
        workspace_name: props.workspaceName,
        region: props.location,
        iterations,
        primary_metric: advancedSettingParams.metric,
        data_script: null,
        compute_target: compute.name,
        task_type: jobType,
        validation_size: advancedSettingParams.validationSize,
        n_cross_validations: advancedSettingParams.nCrossValidations,
        y_min: null,
        y_max: null,
        num_classes: null,
        preprocess: advancedSettingParams.preprocess && advancedSettingParams.jobType !== "forecasting",
        lag_length: 0,
        max_cores_per_iteration: advancedSettingParams.maxCores,
        max_concurrent_iterations: advancedSettingParams.maxConcurrent,
        iteration_timeout_minutes: advancedSettingParams.experimentTimeoutMinutes,
        mem_in_mb: null,
        enforce_time_on_windows: true,
        experiment_timeout_minutes: advancedSettingParams.experimentTimeoutMinutes,
        experiment_exit_score: advancedSettingParams.experimentExitScore,
        whitelist_models: null,
        blacklist_algos: advancedSettingParams.blacklistAlgos,
        auto_blacklist: false,
        blacklist_samples_reached: false,
        exclude_nan_labels: true,
        verbosity: 20,
        debug_log: "automl_errors.log",
        show_warnings: false,
        model_explainability: false,
        service_url: null,
        sdk_url: null,
        sdk_packages: null,
        telemetry_verbosity: "INFO",
        send_telemetry: true,
        spark_service: null,
        metrics: null,
        enable_ensembling: enableEnsembling,
        ensemble_iterations: ensembleIterations,
        enable_tf: false,
        enable_cache: true,
        enable_subsampling: false,
        subsample_seed: null,
        cost_mode: 0,
        metric_operation: MetricOperation[advancedSettingParams.metric],
        is_timeseries: advancedSettingParams.jobType === "forecasting",
        time_column_name: advancedSettingParams.timeSeriesColumn,
        max_horizon: advancedSettingParams.maxHorizon,
        enable_onnx_compatible_models: false,
        enable_voting_ensemble: true,
        enable_stack_ensemble: true,
        enable_stack_ensembling: false,
        grain_column_names: advancedSettingParams.grainColumns,
        enable_early_stopping: false,
        early_stopping_n_iters: 10,
        enable_feature_sweeping: false
    };
};
