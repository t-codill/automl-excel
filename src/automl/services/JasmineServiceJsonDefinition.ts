import { AzureMachineLearningWorkspacesModels } from "@azure/arm-machinelearningservices";
import { defaultSdkVersion } from "../common/defaultSdkVersion";

export type SDKFlight = "default" | "master" | "candidate" | "preview" | "custom";

function getPipDependency(flight: SDKFlight | undefined, sdkVersion: string | undefined): string[] {
    const sdkVersionEquals = sdkVersion ? `==${sdkVersion}` : undefined;
    const noCacheVersion = sdkVersionEquals || `>1.0.0.${Date.now()}`;

    switch (flight) {
        case "master":
            return [
                "--extra-index-url https://azuremlsdktestpypi.azureedge.net/sdk-release/master/588E708E0DF342C4A80BD954289657CF",
                // TODO: Remove the below dataprep*** indexurl once the Feature 422830 is done
                // https://msdata.visualstudio.com/vienna/_workitems/edit/422830/
                // Share Release Pipeline for AzuremlSDK and DataPrep Python SD
                "--extra-index-url https://dataprepdownloads.azureedge.net/pypi/weekly-rc-932B96D048E011E8B56608/latest/",
                `azureml-train-automl${sdkVersionEquals || `<0.1.50.${Date.now()}`}`
            ];
        case "candidate":
            return [
                "--extra-index-url https://azuremlsdktestpypi.azureedge.net/sdk-release/Candidate/604C89A437BA41BD942B4F46D9A3591D",
                `azureml-train-automl${noCacheVersion}`
            ];
        case "preview":
            return [
                "--extra-index-url https://azuremlsdktestpypi.azureedge.net/sdk-release/Preview/E7501C02541B433786111FE8E140CAA1",
                `azureml-train-automl${noCacheVersion}`
            ];
        case "custom":
            return decodeURIComponent(sdkVersion || "")
                .split(";");
        default:
            return [
                `azureml-train-automl${sdkVersionEquals || `==${defaultSdkVersion}`}`
            ];
    }
}

export const getJsonDefinition = (
    compute: AzureMachineLearningWorkspacesModels.ComputeResource,
    flight: SDKFlight | undefined,
    sdkVersion: string | undefined) => {
    return {
        Configuration: {
            script: "train.py",
            arguments: [],
            target: compute.name,
            framework: "python",
            communicator: "None",
            autoPrepareEnvironment: true,
            maxRunDurationSeconds: null,
            nodeCount: 1,
            environment: {
                environmentVariables: {
                    EXAMPLE_ENV_VAR: "EXAMPLE_VALUE"
                },
                python: {
                    userManagedDependencies: false,
                    interpreterPath: "python",
                    condaDependencies: {
                        channels: [
                            "conda-forge"
                        ],
                        name: "project_environment",
                        dependencies: [
                            "python=3.6.2",
                            {
                                pip: getPipDependency(flight, sdkVersion)
                            },
                            "numpy",
                            "py-xgboost<=0.80"
                        ]
                    }
                },
                docker: {
                    enabled: (compute.properties && compute.properties.computeType === "AmlCompute") || false,
                    baseImage: "mcr.microsoft.com/azureml/base:intelmpi2018.3-ubuntu16.04",
                    sharedVolumes: true,
                    gpuSupport: false,
                    shmSize: "1g",
                    arguments: [],
                    baseImageRegistry: {
                        address: null,
                        username: null,
                        password: null
                    }
                },
                spark: {
                    repositories: [],
                    packages: [],
                    precachePackages: true
                },
                databricks: {
                    mavenLibraries: [],
                    pypiLibraries: [],
                    rcranLibraries: [],
                    jarLibraries: [],
                    eggLibraries: []
                }
            },
            history: {
                outputCollection: true,
                snapshotProject: true,
                directoriesToWatch: [
                    "logs"
                ]
            },
            spark: {
                configuration: {
                    "spark.app.name": "Azure ML Experiment",
                    "spark.yarn.maxAppAttempts": 1
                }
            },
            hdi: {
                yarnDeployMode: "cluster"
            },
            tensorflow: {
                workerCount: 1,
                parameterServerCount: 1
            },
            mpi: {
                processCountPerNode: 1
            },
            dataReferences: {},
            sourceDirectoryDataStore: null,
            amlcompute: {
                vmSize: null,
                vmPriority: null,
                retainCluster: false,
                name: null,
                clusterMaxNodeCount: 1
            }
        }
    };
};
