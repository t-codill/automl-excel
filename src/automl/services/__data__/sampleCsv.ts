export const sampleCsv = `a,b,c,d,e
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
`;

const largeCsvHeader = Array(50)
    .fill("aaaaa,bbbbb,ccccc,ddddd,eeeee")
    .join(",");

const largeCsvRow = Array(50)
    .fill("11111,22222,33333,44444,55555")
    .join(",");

export const largeCsv = `${largeCsvHeader}
${Array(50)
        .fill(largeCsvRow)
        .join("\n")}
`;
