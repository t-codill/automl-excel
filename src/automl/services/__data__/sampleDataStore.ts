import { AzureMachineLearningDatastoreManagementModels } from "@vienna/datastore";

export const sampleDataStore: Required<AzureMachineLearningDatastoreManagementModels.DataStoreDto> = {
    dataStoreType: "AzureBlob",
    name: "defaultDataStore",
    azureStorageSection: {
        accountKey: "key",
        accountName: "account",
        credentialType: "AccountKey",
        credential: "secret",
        containerName: "defaultContainer",
        isSas: false
    },
    azureDataLakeSection: {},
    azurePostgreSqlSection: {},
    azureSqlDatabaseSection: {},
    hasBeenValidated: true
};
