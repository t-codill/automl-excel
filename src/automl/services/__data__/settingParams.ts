import { AdvancedSetting } from "../AdvancedSetting";

export const settingParams: AdvancedSetting = {
    jobType: "classification",
    column: "1",
    metric: "AUC_weighted",
    experimentTimeoutMinutes: 60,
    maxIteration: 15,
    experimentExitScore: 0.99,
    maxConcurrent: 1,
    maxCores: 2,
    preprocess: true,
    nCrossValidations: 5,
    blacklistAlgos: null,
    validationSize: 0.2,
    grainColumns: null,
    maxHorizon: null,
    timeSeriesColumn: null
};
