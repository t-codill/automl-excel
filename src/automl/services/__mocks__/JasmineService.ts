import { AzureMachineLearningJasmineManagementModels } from "@vienna/jasmine";

export class JasmineService {

    public async createRun(): Promise<string | undefined> {
        return "AutoML_123";
    }

    public async startRun()
        : Promise<AzureMachineLearningJasmineManagementModels.RunStatus | undefined> {
        return {};
    }

    public async cancelRun(): Promise<{} | undefined> {
        return {};
    }

    public async continueRun(): Promise<{} | undefined> {
        return {};
    }

    public dispose(): void {
        return;
    }

    public reset(): void {
        return;
    }
}
