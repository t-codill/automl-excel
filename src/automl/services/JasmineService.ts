import { AzureMachineLearningWorkspacesModels } from "@azure/arm-machinelearningservices";
import { RestResponse } from "@azure/ms-rest-js";
import { AzureMachineLearningJasmineManagementClient, AzureMachineLearningJasmineManagementModels } from "@vienna/jasmine";
import { ICsvData } from "../common/utils/csv";
import { escapeQuote } from "../common/utils/escapeQuote";
import { stringifyToPython } from "../common/utils/stringifyToPython";
import { AdvancedSetting } from "./AdvancedSetting";
import { getAmlSettings } from "./JasmineServiceAmlSettings";
import { getDataPrepSettings } from "./JasmineServiceDataPrepSettings";
import { getJsonDefinition, SDKFlight } from "./JasmineServiceJsonDefinition";
import { IServiceBaseProps } from "./ServiceBase";
import { ServiceBaseNonArm } from "./ServiceBaseNonArm";

export class JasmineService extends ServiceBaseNonArm<AzureMachineLearningJasmineManagementClient> {
    constructor(props: IServiceBaseProps) {
        super(props, AzureMachineLearningJasmineManagementClient, props.discoverUrls.history);
    }

    public async createRun(
        features: string[],
        previewData: ICsvData,
        experimentName: string,
        compute: AzureMachineLearningWorkspacesModels.ComputeResource,
        dataStoreName: string,
        fileName: string,
        advancedSettingParams: AdvancedSetting): Promise<string | undefined> {

        const response = await this.send(async (client, abortSignal) => {
            return client.jasmine.createParentRun(this.props.subscriptionId, this.props.resourceGroupName, this.props.workspaceName, experimentName, {
                abortSignal,
                timeout: 120000,
                createParentRunDto: this.getRunSettings(
                    features,
                    previewData,
                    experimentName,
                    compute,
                    dataStoreName,
                    fileName,
                    advancedSettingParams)
            });
        });

        if (!response) {
            return undefined;
        }

        return response._response.bodyAsText;
    }

    public async startRun(runId: string, experimentName: string, compute: AzureMachineLearningWorkspacesModels.ComputeResource)
        : Promise<AzureMachineLearningJasmineManagementModels.RunStatus | undefined> {

        const response = this.send(async (client, abortSignal) => {
            return client.jasmine.postRemoteSnapshotRun(this.props.subscriptionId, this.props.resourceGroupName, this.props.workspaceName, experimentName, runId, {
                abortSignal,
                timeout: 120000,
                jsonDefinition: JSON.stringify(getJsonDefinition(compute, this.props.flight.get("SDKFlight") as SDKFlight, this.props.flight.get("SDKVersion")))
            });
        });
        return response;
    }

    public async cancelRun(
        runId: string,
        experimentName: string): Promise<{} | undefined> {

        return this.send(async (client, abortSignal) => {
            return client.jasmine.cancelChildRun(this.props.subscriptionId, this.props.resourceGroupName, this.props.workspaceName, experimentName, runId, {
                abortSignal
            });
        });
    }

    public async continueRun(
        runId: string,
        experimentName: string,
        iterations: number): Promise<RestResponse | undefined> {
        return this.send(async (client, abortSignal) => {
            return client.jasmine.continueRun(this.props.subscriptionId, this.props.resourceGroupName, this.props.workspaceName, experimentName, runId, {
                abortSignal,
                updatedIterations: iterations
            });
        });
    }

    private getRunSettings(
        features: string[],
        previewData: ICsvData,
        experimentName: string,
        compute: AzureMachineLearningWorkspacesModels.ComputeResource,
        dataStoreName: string,
        fileName: string,
        advancedSettingParams: AdvancedSetting): AzureMachineLearningJasmineManagementModels.CreateParentRunDto {

        const primaryMetric = advancedSettingParams.metric;
        const amlSettings = getAmlSettings(experimentName, compute, this.props, advancedSettingParams);
        const dataPrepSettings = getDataPrepSettings(dataStoreName, fileName, previewData, advancedSettingParams.column, features);
        const setting: AzureMachineLearningJasmineManagementModels.CreateParentRunDto = {
            numIterations: advancedSettingParams.maxIteration,
            metrics: [primaryMetric],
            primaryMetric,
            trainSplit: 0,
            maxTimeSeconds: 3600,
            acquisitionParameter: 0,
            numCrossValidation: 5,
            target: compute.name,
            rawAMLSettingsString: stringifyToPython(amlSettings),
            amlSettingsJsonString: JSON.stringify(amlSettings),
            dataPrepJsonString: (escapeQuote(JSON.stringify(dataPrepSettings))),
            enableSubsampling: false
        };
        return setting;
    }
}
