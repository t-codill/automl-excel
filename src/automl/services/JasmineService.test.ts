import { testContext } from "../common/context/__data__/testContext";
import { settingParams } from "./__data__/settingParams";
import { JasmineService } from "./JasmineService";
import * as JasmineServiceAmlSettings from "./JasmineServiceAmlSettings";
import * as JasmineServiceJsonDefinition from "./JasmineServiceJsonDefinition";

jest.mock("@vienna/jasmine");

const service = new JasmineService(testContext);

describe("JasmineService", () => {
    beforeEach(() => {
        const jsonSpy: jest.SpyInstance = jest.spyOn(JasmineServiceJsonDefinition, "getJsonDefinition");
        jsonSpy.mockReturnValue("##JasmineServiceJsonDefinition##");
        const amlSpy: jest.SpyInstance = jest.spyOn(JasmineServiceAmlSettings, "getAmlSettings");
        amlSpy.mockReturnValue("##JasmineServiceAmlSettings##");
    });
    it("should createRun", async () => {
        const result = await service.createRun(
            ["feature1", "feature2"],
            {
                data: [],
                delimiter: ",",
                hasHeader: true,
                header: []
            },
            "sampleExperimentName",
            { name: "sampleCompute", properties: { computeType: "VirtualMachine" } },
            "sampleDataStore",
            "sampleFileName",
            settingParams);
        expect(result)
            .toMatchSnapshot();
    });

    it("should startRun", async () => {
        const result = await service.startRun("AutoML_000", "sampleExperiment", { name: "sampleCompute", properties: { computeType: "VirtualMachine" } });
        expect(result)
            .toMatchSnapshot();
    });

    it("should cancelRun", async () => {
        const result = await service.cancelRun("AutoML_000", "sampleExperiment");
        expect(result)
            .toMatchSnapshot();
    });

    it("should continueRun", async () => {
        const result = await service.continueRun("AutoML_000", "sampleExperiment", 100);
        expect(result)
            .toMatchSnapshot();
    });

    it("should return undefined for disposed service", async () => {
        const result = service.createRun(
            ["feature1", "feature2"],
            {
                data: [],
                delimiter: ",",
                hasHeader: true,
                header: []
            },
            "sampleExperimentName",
            { name: "sampleCompute", properties: { computeType: "VirtualMachine" } },
            "sampleDataStore",
            "sampleFileName",
            settingParams);
        service.dispose();
        expect(await result)
            .toBeUndefined();
    });
});
