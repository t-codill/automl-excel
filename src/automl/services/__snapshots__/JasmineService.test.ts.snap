// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JasmineService should cancelRun 1`] = `
Object {
  "_mockParameters": Array [
    "00000000-0000-0000-0000-000000000000",
    "testResource",
    "testWorkSpace",
    "sampleExperiment",
    "AutoML_000",
    Object {
      "abortSignal": Object {},
    },
  ],
  "_response": Object {
    "headers": HttpHeaders {
      "_headersMap": Object {},
    },
    "request": WebResource {
      "abortSignal": undefined,
      "body": undefined,
      "formData": undefined,
      "headers": HttpHeaders {
        "_headersMap": Object {},
      },
      "method": "GET",
      "onDownloadProgress": undefined,
      "onUploadProgress": undefined,
      "proxySettings": undefined,
      "query": undefined,
      "streamResponseBody": undefined,
      "timeout": 0,
      "url": "",
      "withCredentials": false,
    },
    "status": 200,
  },
}
`;

exports[`JasmineService should continueRun 1`] = `
Object {
  "_mockParameters": Array [
    "00000000-0000-0000-0000-000000000000",
    "testResource",
    "testWorkSpace",
    "sampleExperiment",
    "AutoML_000",
    Object {
      "abortSignal": Object {},
      "updatedIterations": 100,
    },
  ],
  "_response": Object {
    "headers": HttpHeaders {
      "_headersMap": Object {},
    },
    "request": WebResource {
      "abortSignal": undefined,
      "body": undefined,
      "formData": undefined,
      "headers": HttpHeaders {
        "_headersMap": Object {},
      },
      "method": "GET",
      "onDownloadProgress": undefined,
      "onUploadProgress": undefined,
      "proxySettings": undefined,
      "query": undefined,
      "streamResponseBody": undefined,
      "timeout": 0,
      "url": "",
      "withCredentials": false,
    },
    "status": 200,
  },
}
`;

exports[`JasmineService should createRun 1`] = `"[\\"00000000-0000-0000-0000-000000000000\\",\\"testResource\\",\\"testWorkSpace\\",\\"sampleExperimentName\\",{\\"abortSignal\\":{},\\"timeout\\":120000,\\"createParentRunDto\\":{\\"numIterations\\":15,\\"metrics\\":[\\"AUC_weighted\\"],\\"primaryMetric\\":\\"AUC_weighted\\",\\"trainSplit\\":0,\\"maxTimeSeconds\\":3600,\\"acquisitionParameter\\":0,\\"numCrossValidation\\":5,\\"target\\":\\"sampleCompute\\",\\"rawAMLSettingsString\\":\\"\\\\\\"##JasmineServiceAmlSettings##'\\",\\"amlSettingsJsonString\\":\\"\\\\\\"##JasmineServiceAmlSettings##\\\\\\"\\",\\"dataPrepJsonString\\":\\"{\\\\\\\\\\\\\\"datastoreName\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"sampleDataStore\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"dataPath\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"sampleFileName\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"columnSeparator\\\\\\\\\\\\\\":\\\\\\\\\\\\\\",\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"promoteHeader\\\\\\\\\\\\\\":true,\\\\\\\\\\\\\\"encoding\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"UTF8\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"ignoreNewlineInQuotes\\\\\\\\\\\\\\":true,\\\\\\\\\\\\\\"skipRows\\\\\\\\\\\\\\":0,\\\\\\\\\\\\\\"features\\\\\\\\\\\\\\":[\\\\\\\\\\\\\\"feature1\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"feature2\\\\\\\\\\\\\\"],\\\\\\\\\\\\\\"label\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"1\\\\\\\\\\\\\\"}\\",\\"enableSubsampling\\":false}}]"`;

exports[`JasmineService should startRun 1`] = `
Object {
  "_mockParameters": Array [
    "00000000-0000-0000-0000-000000000000",
    "testResource",
    "testWorkSpace",
    "sampleExperiment",
    "AutoML_000",
    Object {
      "abortSignal": Object {},
      "jsonDefinition": "\\"##JasmineServiceJsonDefinition##\\"",
      "timeout": 120000,
    },
  ],
  "runId": undefined,
}
`;
